packages/dds/tree/src/test/simple-tree/api/schemaCreationUtilities.spec.ts

```diff
@@ -235,4 +235,99 @@ describe("schemaCreationUtilities", () => {
 			const _b: InsertableImplicit = new DayNodes.Tomorrow();
 		}
 	});
+
+	// Highest priority issue
+	it("enum interop - adaptEnum with explicit schema", () => {
+		enum Day {
+			Today = "today",
+			Tomorrow = "tomorrow",
+		}
+
+		function getDay(value: string): Day {
+			switch (value) {
+				case "today":
+					return Day.Today;
+				case "tomorrow":
+					return Day.Tomorrow;
+				default:
+					throw new Error(`Invalid value: ${value}`);
+			}
+		}
+
+		const DayNodes = adaptEnum(schema, Day);
+
+		class DayObject extends schema.object("DayObject", {
+			value: DayNodes.schema,
+			name: schema.string,
+		}) {}
+		function createDayObjects(value: Day, name: string): DayObject[] {
+			// @ts-expect-error Day is not compatible with value of Day.Today
+			const dayObject1 = new DayObject({ value: DayNodes("today"), name: "1" });
```

Comment:
If you want to work with string unions instead of enum values, this works fine:

```typescript
	it("enumFromStrings", () => {
		const days = ["today", "tomorrow"] as const;
		type Day = (typeof days)[number];
		const DayNodes = enumFromStrings(schema, days);

		class DayObject extends schema.object("DayObject", {
			value: DayNodes.schema,
			name: schema.string,
		}) {}
		function createDayObjects(value: Day, name: string): DayObject[] {
			const dayObject1 = new DayObject({ value: DayNodes("today"), name: "1" });
			const dayObject2 = new DayObject({ value: DayNodes("today" as Day), name: "2" });
			return [dayObject1, dayObject2];
		}
	});
```

If you want to work with enum values, you need to convert to an enum value and not use strings since typescript adds extra type safety and does not let you intermix the two.

Reply:
Thanks!